src/main/java/uk/ac/soton/itinnovation/security/systemmodeller/rest/EntityController.java

```diff
@@ -0,0 +1,835 @@
+import uk.ac.soton.itinnovation.security.modelquerier.dto.MisbehaviourSetDB;
+import uk.ac.soton.itinnovation.security.modelquerier.dto.ThreatDB;
+import uk.ac.soton.itinnovation.security.modelquerier.dto.TrustworthinessAttributeSetDB;
+import uk.ac.soton.itinnovation.security.semanticstore.AStoreWrapper;
+import uk.ac.soton.itinnovation.security.semanticstore.JenaTDBStoreWrapper;
+import uk.ac.soton.itinnovation.security.systemmodeller.model.Model;
+import uk.ac.soton.itinnovation.security.systemmodeller.model.WebKeyRole;
+import uk.ac.soton.itinnovation.security.systemmodeller.rest.exceptions.InternalServerErrorException;
+import uk.ac.soton.itinnovation.security.systemmodeller.semantics.StoreModelManager;
+import uk.ac.soton.itinnovation.security.systemmodeller.util.SecureUrlHelper;
+
+/**
+ * Includes all operations of the Entity Controller Service.
+ */
+@RestController
+public class EntityController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private StoreModelManager storeModelManager;
+
+    @Autowired
+    private SecureUrlHelper secureUrlHelper;
+
+    @Value("${admin-role}")
+    public String adminRole;
+
+    /**
+     * This REST method retrieves a JSON document describing a specific system model threat.
+     *
+     * @param modelId the String representation of the model object to search
+     * @param uri     threat URI (of the form "system#ThreatName_ID")
+     * @return A JSON representation of a threat object
+     * @throws InternalServerErrorException if an error occurs during report generation
+     */
+    @RequestMapping(value = "/models/{modelId}/entity/system/threats/{uri}", method = RequestMethod.GET)
+    public ResponseEntity<ThreatDB> getEntitySystemThreat(@PathVariable String modelId, @PathVariable String uri) {
+
+        logger.info("get system threat for model {} with URI: {}", modelId, uri);
+
+        final Model model = secureUrlHelper.getModelFromUrlThrowingException(modelId, WebKeyRole.READ);
+
+        AStoreWrapper store = storeModelManager.getStore();
+
+        try {
+            logger.info("Initialising JenaQuerierDB");
+
+            JenaQuerierDB querierDB = new JenaQuerierDB(((JenaTDBStoreWrapper) store).getDataset(),
+                    model.getModelStack(), false);
+
+            querierDB.init();
+
+            logger.info("getting threat");
+
+            ThreatDB threat = querierDB.getThreat(uri, "system-inf");
+
+            if (threat == null) {
+                return ResponseEntity.notFound().build();
+            }
+
+            return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(threat);
+
+        } catch (Exception e) {
+            logger.error("Simple API get threat failed due to an error", e);
+            throw new InternalServerErrorException(
+                    "Threat fetch failed. Please contact support for further assistance.");
+        }
+    }
+
+    /**
+     * This REST method ...
```

Comment:
"This REST method" does what?

Comment applies to all methods

Reply:
I've added short descriptions for all methods address @kenmeacham comments 1,2,3, and 5